### PR TITLE
Ako/ Add language switcher

### PR DESCRIPTION
**Pull request by ali-hosseini-deriv**
This is to add language switcher component.
The desktop mode can be visible in storybook and the mobile is implemented as part of playground.

### CODE REVIEW

File: src/components/AppLayout/Drawer/DrawerHeader/DrawerHeader.scss [hunk 1]
```diff
@@ -19,5 +19,6 @@
         flex: 1;
         overflow-y: auto;
         align-items: center;
+        justify-content: space-between;
```
Comment (niloofar-deriv) [inline src/components/AppLayout/Drawer/DrawerHeader/DrawerHeader.scss:22]: Do we need this?

File: src/main.ts [hunk 2]
```diff
@@ -35,3 +35,5 @@ export {
     PlatformSwitcherItem,
 } from "./components/AppLayout";
 export { ContextMenu } from "./components/ContextMenu";
+export { DesktopLanguagesModal } from "./components/AppLayout/LanguagesModal/DesktopLanguagesModal";
```
Comment (niloofar-deriv) [inline src/main.ts:38]: Please export these in `src/components/AppLayout/index.ts`

```
export { DesktopLanguagesModal } from "./LanguagesModal/DesktopLanguagesModal";
export { MobileLanguagesDrawer } from "./LanguagesModal/MobileLanguagesDrawer";
```

File: src/main.ts [hunk 3]
```diff
@@ -35,3 +35,5 @@ export {
     PlatformSwitcherItem,
```
Comment (niloofar-deriv) [inline src/main.ts:35]:
```suggestion
    PlatformSwitcherItem,
    DesktopLanguagesModal,
    MobileLanguagesDrawer
```

File: src/components/AppLayout/LanguagesModal/MobileLanguagesDrawer.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            {isOpen &&
+                languages.map((language) => {
+                    const Icon = () => language.icon;
+                    const onButtonClick = () => {
+                        onLanguageSwitch(language.code);
+                        onClose();
+                    };
+                    return (
+                        <div key={language.code} className="languages-body__item">
+                            <Button
```
Comment (niloofar-deriv) [inline src/components/AppLayout/LanguagesModal/MobileLanguagesDrawer.tsx:45]: Please replace this with <LanguageItem /> component

File: src/components/AppLayout/Drawer/DrawerHeader/DrawerHeader.scss [hunk 5]
```diff
@@ -19,5 +19,6 @@
...
     }
 }
```
Comment (niloofar-deriv) [inline src/components/AppLayout/Drawer/DrawerHeader/DrawerHeader.scss:24]:
```suggestion
}

```

File: src/components/AppLayout/LanguagesModal/LanguageItem.tsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    currentLang: string;
+};
+export const LanguageItem = ({
+    language,
+    onButtonClick,
+    currentLang,
+}: TLangComponent) => (
+    <Button
+        className="languages__body-button"
+        color="black"
```
Comment (niloofar-deriv) [inline src/components/AppLayout/LanguagesModal/LanguageItem.tsx:22]: I guess this is redundant 
```suggestion
       
```

File: src/components/AppLayout/Drawer/DrawerHeader/DrawerHeader.scss (hunk 1 above)
Comment (ali-hosseini-deriv) [reply src/components/AppLayout/Drawer/DrawerHeader/DrawerHeader.scss:22]: yess that's for drawer header items to be aligned

File: src/components/AppLayout/LanguagesModal/DesktopLanguagesModal.tsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
+import React, { ReactElement } from "react";
+import { Modal, Text } from "../../../main";
+import "./LanguagesModal.scss";
+import { IconTypes } from "@deriv/quill-icons";
+import { LanguageItem } from "./LanguageItem";
+
+type TLanguagesModalProps = {
+    isModalOpen: boolean;
+    languages: {
```
Comment (niloofar-deriv) [inline src/components/AppLayout/LanguagesModal/DesktopLanguagesModal.tsx:9]:
```suggestion
    languages: ComponentProps<typeof LanguageItem>['language'][]
```

File: src/components/AppLayout/LanguagesModal/LanguageItem.tsx [hunk 8]
```diff
@@ -0,0 +1,34 @@
+import { ReactElement } from "react";
+import { Button } from "../../Button";
+import { Text } from "../../Text";
+import { IconTypes } from "@deriv/quill-icons";
+type TLanguageItem = {
+    code: string;
+    display_name: string | ReactElement;
```
Comment (niloofar-deriv) [inline src/components/AppLayout/LanguagesModal/LanguageItem.tsx:7]:
```suggestion
    displayName: string;
```

File: src/components/AppLayout/LanguagesModal/LanguageItem.tsx [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+    icon: IconTypes;
```
Comment (niloofar-deriv) [inline src/components/AppLayout/LanguagesModal/LanguageItem.tsx:8]:
```suggestion
    Icon: ReactNode;
```

Comment (niloofar-deriv) [reply src/components/AppLayout/LanguagesModal/LanguageItem.tsx:8]: Just go inside Button component and add this 😁 
`icon?: ReactElement | ReactNode;`

it dose not break anything believe me 😬

File: src/components/AppLayout/LanguagesModal/LanguageItem.tsx [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    onButtonClick: () => void;
+    currentLang: string;
+};
+export const LanguageItem = ({
+    language,
+    onButtonClick,
+    currentLang,
+}: TLangComponent) => (
+    <Button
+        className="languages__body-button"
```
Comment (niloofar-deriv) [inline src/components/AppLayout/LanguagesModal/LanguageItem.tsx:21]:
```suggestion
        className="languages-body__button"
```

File: src/components/AppLayout/LanguagesModal/MobileLanguagesDrawer.tsx (hunk 4 above)
Comment (ali-hosseini-deriv) [reply src/components/AppLayout/LanguagesModal/MobileLanguagesDrawer.tsx]: done

File: src/main.ts (hunk 3 above)
Comment (ali-hosseini-deriv) [reply src/main.ts:35]: done

File: src/main.ts (hunk 2 above)
Comment (ali-hosseini-deriv) [reply src/main.ts]: done

File: src/components/AppLayout/Drawer/DrawerHeader/DrawerHeader.scss (hunk 5 above)
Comment (ali-hosseini-deriv) [reply src/components/AppLayout/Drawer/DrawerHeader/DrawerHeader.scss]: done

File: src/components/AppLayout/LanguagesModal/LanguageItem.tsx (hunk 6 above)
Comment (ali-hosseini-deriv) [reply src/components/AppLayout/LanguagesModal/LanguageItem.tsx]: done

File: src/components/AppLayout/LanguagesModal/LanguageItem.tsx (hunk 9 above)
Comment (ali-hosseini-deriv) [reply src/components/AppLayout/LanguagesModal/LanguageItem.tsx]: sure :-D 
done

File: src/components/AppLayout/LanguagesModal/LanguageItem.tsx (hunk 8 above)
Comment (ali-hosseini-deriv) [reply src/components/AppLayout/LanguagesModal/LanguageItem.tsx]: done

File: playground/index.tsx [hunk 11]
```diff
@@ -90,7 +91,14 @@ AccountSwitcher.setAppElement(document.getElementById("root") as HTMLElement);
 const App = () => {
     const { isMobile } = useDevice();
     const [is_drawer_open, setDrawerOpen] = React.useState(false);
-
+    const [selectedLang, setSelectedLang] = React.useState("en");
```
Comment (prince-deriv) [inline playground/index.tsx:94]:
```suggestion
    const [selectedLanguage, setSelectedLanguage] = React.useState("en");
```

File: playground/index.tsx [hunk 12]
```diff
@@ -90,7 +91,14 @@ AccountSwitcher.setAppElement(document.getElementById("root") as HTMLElement);
...
+    const [isLanguageDrawerOpen, setIsLanguagesDrawerOpen] =
```
Comment (prince-deriv) [inline playground/index.tsx:95]:
```suggestion
    const [isLanguagesDrawerOpen, setIsLanguagesDrawerOpen] =
```

File: playground/index.tsx [hunk 13]
```diff
@@ -90,7 +91,14 @@ AccountSwitcher.setAppElement(document.getElementById("root") as HTMLElement);
...
+        React.useState(false);
+    const SelectedLanguageIcon =
+        LanguagesItemsDrawerConfig.languages.find((language) => {
+            return language.code === selectedLang;
+        })?.icon ?? LanguagesItemsDrawerConfig.languages[0].icon;
+    console.log(SelectedLanguageIcon);
```
Comment (prince-deriv) [inline playground/index.tsx:101]: Do we still need to console?

**Review by prince-deriv — CHANGES_REQUESTED** [review]

**Review by niloofar-deriv — APPROVED** [review]
LGTM 👍

File: playground/index.tsx (hunk 13 above)
Comment (niloofar-deriv) [reply playground/index.tsx:101]: @prince-deriv the playground file is not a parts of build and it's just a test file for ourselves so it dose not matter how we are naming variable and states and the rest of the things

Comment (ali-hosseini-deriv) [reply playground/index.tsx]: that's right, anyways I've done the suggested changes

File: playground/index.tsx (hunk 12 above)
Comment (ali-hosseini-deriv) [reply playground/index.tsx]: done

**Review by niloofar-deriv — APPROVED** [review]

**Review by prince-deriv — APPROVED** [review]

**Review by shayan-deriv — APPROVED** [review]

**Pull request merged**